tools/kvmexit.py

```diff
@@ -0,0 +1,293 @@
+#!/usr/bin/env python
+#
+# kvmexit.py
+#
+# Display the exit_reason and its statistics of each kvm_exit
+# for all vcpus of all virtual machines. For example,
+# $./kvmexit.py
+#  TGID     PID      KVM_EXIT_REASON                     STAT
+#  1273551  1273568  EXIT_REASON_MSR_WRITE               6
+#  1274253  1274261  EXIT_REASON_EXTERNAL_INTERRUPT      1
+#  1274253  1274261  EXIT_REASON_HLT                     12
+#  ...
+#
+# @TGID: each vitual machine's pid in the user space.
+# @PID: the user space's thread of each vcpu of that virtual machine.
+# @KVM_EXIT_REASON: the reason why the vm exits.
+# @STAT: the counts of the @KVM_EXIT_REASON since the vm starts, and
+#        it is recorded every second by setting `sleep(1)`.
+#
+# REQUIRES: Linux 4.7+ (BPF_PROG_TYPE_TRACEPOINT support)
+#
+# Copyright (c) 2019 BYTEDANCE Inc.
+#
+# Author(s):
+#   Fei Li <lifei.shirley@bytedance.com>
+
+
+from __future__ import print_function
+from time import sleep, strftime
+from bcc import BPF
+import argparse
+import multiprocessing
+import signal
+import subprocess
+
+#
+# Process Arguments
+#
+
+# arg validation
+def positive_int(val):
+    try:
+        ival = int(val)
+    except ValueError:
+        raise argparse.ArgumentTypeError("must be an integer")
+
+    if ival < 0:
+        raise argparse.ArgumentTypeError("must be positive")
+    return ival
+
+def positive_nonzero_int(val):
+    ival = positive_int(val)
+    if ival == 0:
+        raise argparse.ArgumentTypeError("must be nonzero")
+    return ival
+
+# arguments
+examples = """examples:
+    ./kvmexit             # Display kvm_exit_reason and its statistics until Ctrl-C
+    ./kvmexit 5           # Display kvm_exit_reason and its statistics for 5 seconds only
+"""
+parser = argparse.ArgumentParser(
+    description="Display kvm_exit_reason and its statistics at a timed interval",
+    formatter_class=argparse.RawDescriptionHelpFormatter,
+    epilog=examples)
+parser.add_argument("duration", nargs="?", default=99999999,
+    type=positive_nonzero_int,
```

Comment:
Just use "type=int" and we are done with it. Similar to most of other tools, no need to validate actual value. Tools users are supposed to look at example/man page and provides a sane input.

Reply:
> Just use "type=int" and we are done with it. Similar to most of other tools, no need to validate actual value. Tools users are supposed to look at example/man page and provides a sane input.

Got it, thanks for the reminder :)